### PR TITLE
Md docs 11665

### PR DESCRIPTION
**Pull request by md3822**

### CODE REVIEW

File: AggregateData/README.md [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+						      Arrays.asList(
+						              Aggregates.match(Filters.eq("shippingAddress.state", state)),
+						              Aggregates.group("$shippingAddress.state", Accumulators.avg("Average Shipping", "$shipping"))
+						      )
+						).into(new ArrayList<Document>());
+					return queryResult;
+```
+
+This method illustrates Average. The example uses Aggregates.group to build the $group stage and Accumulators.avg to build the accumulator expression. 
+* A $match stage to filter for documents whose *shippingAdress - state* contains the parameter passed.
```
Comment (Raghav2018) [inline AggregateData/README.md:110]: typo

File: AggregateData/testsrc/AggregateDataTest.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	public void TestConnection() {		
+		MongoClientURI clientUri = new MongoClientURI(CONNECTION_STRING);
+		try(MongoClient client = new MongoClient(clientUri))
+		{
+			assertNotNull(client);
+		}
+	}
+		
+		@Test
+		public void AggregateByStateTest() {
```
Comment (Raghav2018) [inline AggregateData/testsrc/AggregateDataTest.java]: naming convention is incorrect. MethodYouAreTesting_Scenario_ExpectedResult()

File: AggregateData/testsrc/AggregateDataTest.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+		
+		@Test(expected=IllegalArgumentException.class)
+		public void AggregateByStateTest_InvalidState_ThrowsException() {
+			
+			AggregateData aggregateData = new AggregateData();	
+			List<Document> queryResult = aggregateData.AggregateByState(CONNECTION_STRING, "");    
+		}
+		
+		@Test
+		public void AggregateByStateTest_ValidArguments_SuccessWithNoRecords_ThrowsException() {
```
Comment (Raghav2018) [inline AggregateData/testsrc/AggregateDataTest.java]: Wrong naming convention again

File: AggregateData/testsrc/AggregateDataTest.java [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+			AggregateData aggregateData = new AggregateData();	
+			List<Document> queryResult = aggregateData.AggregateByState(CONNECTION_STRING, "Quebec");
+			
+			
+			//verify size
+		    assertEquals(0, queryResult.size());  
+		}
+		
+		@Test
+		public void AggregateByStateAverageShippingTest() {
```
Comment (Raghav2018) [inline AggregateData/testsrc/AggregateDataTest.java]: Naming convention is incorrect

File: AggregateData/testsrc/AggregateDataTest.java [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+		
+		@Test(expected=IllegalArgumentException.class)
+		public void AggregateByStateAverageShippingTest_InvalidState_ThrowsException() {
+			
+			AggregateData aggregateData = new AggregateData();	
+			List<Document> queryResult = aggregateData.AggregateByState(CONNECTION_STRING, "");    
+		}
+		
+		@Test
+		public void AggregateByStateAverageShippingTest_ValidArguments_SuccessWithNoRecords_ThrowsException() {
```
Comment (Raghav2018) [inline AggregateData/testsrc/AggregateDataTest.java]: Naming convention is incorrect

File: AggregateData/testsrc/AggregateDataTest.java [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+			AggregateData aggregateData = new AggregateData();	
+			List<Document> queryResult = aggregateData.AggregateByState(CONNECTION_STRING, "Quebec"); 
+			
+			
+			//verify size
+		    assertEquals(0, queryResult.size());  
+		}
+		
+		@Test
+		public void AggregateByCityCheckStatusTest() {
```
Comment (Raghav2018) [inline AggregateData/testsrc/AggregateDataTest.java]: Naming convention is incorrect

File: AggregateData/src/AggregateData.java [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.Projections;
+import com.mongodb.client.model.Aggregates;
+import com.mongodb.client.model.Accumulators;
+import com.mongodb.client.model.Filters;
+
+public class AggregateData {
+		
+		public List<Document> AggregateByState(String connectionString, String state){
```
Comment (Raghav2018) [inline AggregateData/src/AggregateData.java]: Please add meta data explaining the methods

File: AggregateData/README.md [hunk 8]
```diff
@@ -0,0 +1,390 @@
...
+
+This is a sample application that demonstrates MongoDB Java Driver's Aggregation Framework. It has three methods:
+
+* one that illustrates count
+* one that illustrates average
+* one that illustrates projection
+
+## Getting Started
+
+The sample application demonstrates deletion of data from a MongoDB Atlas free-tier cluster. The cluster contains stores.orders collection which mimics an e-commerce database. The general schema is as follows:
```
Comment (Raghav2018) [inline AggregateData/README.md:13]: Deletion of data? Really?

**Review by Raghav2018 — COMMENTED** [review]
I have put up my review comments. Please take a look

**Comment by md3822** [comment]
I have made the changes accordingly. Please review

**Comment by Raghav2018** [comment]
LGTM

**Pull request merged**